### PR TITLE
fix(Compiler): fix text nodes after content tags

### PR DESCRIPTION
**Pull request by vicb**
Before 

```
<content></content>
{{ }} // text node, index = 1

 // compiles to

<script start></script> // a single <content> generates 2 <script> -> mess with offset
<script end></script>
{{ }} // text node, index = 1 the actual index is 2
```

After

```
<content></content>
{{ }} // text node, index = 1

 // compiles to

<script start></script> // the end script marker is generated lazily when needed
{{ }} // text node, index = 1
```

/cc @vsavkin @yjbanov

### CODE REVIEW

**Comment by vicb** [comment]
Added a last commit to test all 3 strategies - before only the last available strategy would be tested: `beforeEachBindings()` was called before the describe, ie called several time in a row and only the last override would be active.

File: modules/angular2/test/render/dom/shadow_dom_emulation_integration_spec.ts [hunk 1]
```diff
@@ -56,10 +61,22 @@ export function main() {
     }
 
     StringMapWrapper.forEach(strategies, (strategyBinding, name) => {
+      describe(`${name} shadow dom strategy`, () => {
+        beforeEachBindings(() => { return [strategyBinding, DomTestbed]; });
 
-      beforeEachBindings(() => { return [strategyBinding, DomTestbed]; });
+        // GH-2095 - https://github.com/angular/angular/issues/2095
+        it('should support text nodes after content tags',
```
Comment (tbosch) [inline modules/angular2/test/render/dom/shadow_dom_emulation_integration_spec.ts]: Could you rewrite this test to not use the `TestComponentBuilder`, but only the `DomTestBed`?

Comment (tbosch) [inline modules/angular2/test/render/dom/shadow_dom_emulation_integration_spec.ts]: As this test is part of the render side, I don't want it to use the application compiler...

**Comment by vicb** [comment]
@tbosch I have updated the test to use the `DomTestBed`, that was actually pretty easy thanks to your hints - I guess I failed earlier because of the bug in the tests (they were testing the native strategy 3 times).

You also mentioned a corner case ?

**Comment by tbosch** [comment]
@vicb sorry, false alarm. This PR is good to go.

File: modules/angular2/test/render/dom/shadow_dom_emulation_integration_spec.ts [hunk 2]
```diff
@@ -493,3 +516,9 @@ var tabTemplate = new ViewDefinition({
   template: '<div><div *auto="cond">TAB(<content></content>)</div></div>',
   directives: [autoViewportDirective]
 });
+
+@Component({selector: 'textAfterContent'})
```
Comment (tbosch) [inline modules/angular2/test/render/dom/shadow_dom_emulation_integration_spec.ts]: You don't need these annotations for the test...

**Comment by tbosch** [comment]
Besides the comment above :-)

**Comment by vicb** [comment]
landed as 6e38515402c5763a0b41e66da129e45b0be80208 & previous commits
